CMakeLists.txt

```diff
@@ -59,3 +60,37 @@ if (UTF8PROC_INSTALL)
       "${PROJECT_SOURCE_DIR}/utf8proc.h"
     DESTINATION include)
 endif()
+
+if(UTF8PROC_ENABLE_TESTING)
+  enable_testing()
+  add_executable(case test/tests.h test/tests.c utf8proc.h test/case.c)
+  target_link_libraries(case utf8proc)
+  add_executable(custom test/tests.h test/tests.c utf8proc.h test/custom.c)
+  target_link_libraries(custom utf8proc)
+  add_executable(iterate test/tests.h test/tests.c utf8proc.h test/iterate.c)
+  target_link_libraries(iterate utf8proc)
+  add_executable(misc test/tests.h test/tests.c utf8proc.h test/misc.c)
+  target_link_libraries(misc utf8proc)
+  add_executable(printproperty test/tests.h test/tests.c utf8proc.h test/printproperty.c)
+  target_link_libraries(printproperty utf8proc)
+  add_executable(valid test/tests.h test/tests.c utf8proc.h test/valid.c)
+  target_link_libraries(valid utf8proc)
+  add_test(testcase case)
+  add_test(testcustom custom)
+  add_test(testiterate iterate)
+  add_test(testmisc misc)
+  add_test(testprintproperty printproperty)
+  add_test(testvalid valid)
+  if (NOT WIN32)
+    # Some test disabled, because they don't compile on Windows, missing getline, ...
+    add_executable(charwidth test/tests.h test/tests.c utf8proc.h test/charwidth.c)
+    target_link_libraries(charwidth utf8proc)
+    #add_executable(graphemetest test/tests.h test/tests.c utf8proc.h test/graphemetest.c)
+    #target_link_libraries(graphemetest utf8proc)
+    #add_executable(normtest test/tests.h test/tests.c utf8proc.h test/normtest.c)
+    #target_link_libraries(normtest utf8proc)
+    add_test(testcharwidth charwidth)
+    # add_test(testgraphemetest graphemetest GraphemeBreakTest.txt)
+    # add_test(testnormtest normtest)
```

Comment:
Why are these commented out?   You call them as
```
	test/normtest data/NormalizationTest.txt
	test/graphemetest data/GraphemeBreakTest.txt
```
but then you need a makefile rule to download these two data files.

Reply:
It's commented out because the .txt files are missing, because in CMakeLists.txt is missing building these.